client/src/types/stronghold.rs

```diff
@@ -230,6 +234,32 @@ impl Stronghold {
         Ok(snapshot)
     }
 
+    /// Stores the key to write to the [`Snapshot`] at [`Location`]. This operation zeroizes the key
+    /// after successful insertion
+    pub fn store_snapshot_key_at_location<K>(&self, mut key: K, location: Location) -> Result<(), ClientError>
+    where
+        K: AsMut<[u8]> + Zeroize,
+    {
+        let key = key.as_mut();
+        // magic constant!
+        if key.len() != 32 {
+            return Err(ClientError::IllegalKeySize(key.len()));
+        }
+
+        let mut key_location = self.key_location.write().map_err(|e| ClientError::LockAcquireFailed)?;
+        key_location.replace(location.clone());
+
+        let (vault_id, record_id) = location.resolve();
+
+        let mut snapshot = self.get_snapshot()?;
+        let mut kkey = [0u8; 32];
+        kkey.copy_from_slice(key);
+        key.zeroize();
+        snapshot.store_snapshot_key(kkey, vault_id, record_id)?;
```

Comment:
Should inside of `store_snapshot_key()` also be `snapshot_key.zeroize();` at the end?

Reply:
yes, it should. good catch, but I will exchange the appearance of any old `Key` generics to the `KeyProvider` to keep it consistent. I think that's the way to go.